eth/blockwatch/client.go

```diff
@@ -79,31 +85,24 @@ func (rc *RPCClient) HeaderByNumber(number *big.Int) (*MiniHeader, error) {
 		return nil, ethereum.NotFound
 	}
 
+	// If no L1BlockNumber, then Livepeer is running on L1, so L1BlockNumber is the same as BlockNumber
+	if header.L1BlockNumber == "" {
+		header.L1BlockNumber = header.Number
+	}
+
 	blockNum, ok := math.ParseBig256(header.Number)
 	if !ok {
 		return nil, errors.New("Failed to parse big.Int value from hex-encoded block number returned from eth_getBlockByNumber")
 	}
-	miniHeader := &MiniHeader{
-		Hash:   header.Hash,
-		Parent: header.ParentHash,
-		Number: blockNum,
-	}
-	return miniHeader, nil
-}
-
-// HeaderByHash fetches a block header by its block hash. If no block exists with this number it will return
-// a `ethereum.NotFound` error.
-func (rc *RPCClient) HeaderByHash(hash common.Hash) (*MiniHeader, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), rc.requestTimeout)
-	defer cancel()
-	header, err := rc.client.HeaderByHash(ctx, hash)
-	if err != nil {
-		return nil, err
+	l1BlockNum, ok := math.ParseBig256(header.L1BlockNumber)
+	if !ok {
+		return nil, errors.New("Failed to parse big.Int value from hex-encoded L1 block number returned from eth_getBlockByNumber")
```

Comment:
```suggestion
		return nil, fmt.Errorf("Failed to parse big.Int value from hex-encoded L1 block number returned from %v", method)
```

since the method name is a variable.